schema/auth.json

```diff
@@ -0,0 +1,214 @@
+{
+  "$id": "https://serverlessworkflow.io/schemas/0.7/auth.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "description": "Serverless Workflow specification - auth schema",
+  "type": "object",
+  "auth": {
+    "oneOf": [
+      {
+        "type": "string",
+        "format": "uri",
+        "description": "URI to a resource containing auth definitions (json or yaml)"
+      },
+      {
+        "type": "array",
+        "description": "Workflow auth definitions",
+        "items": {
+          "type": "object",
+          "$ref": "#/definitions/authdef"
+        },
+        "additionalItems": false,
+        "minItems": 1
+      }
+    ]
+  },
+  "required": [
+    "auth"
+  ],
+  "definitions": {
+    "authdef": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "Unique auth name",
+          "minLength": 1
+        },
+        "scheme": {
+          "type": "string",
+          "description": "Defines the auth type",
+          "enum": [
+            "basic",
+            "bearer",
+            "oidc"
```

Comment:
You will hate me, but `OIDC` is not the right term, in the end. That became obvious to me when reading the [token exchange RFC](https://datatracker.ietf.org/doc/html/rfc8693), where it is clearly stated that token exchange is an `oauth2` grant type.
So yeah, we should replace `oidc` by `oauth2`.
So sorry about that, it's already the second time I'm asking to rename those 😒

Reply:
updated